### DISCUSSION TITLE
power consumption

### DISCUSSION THREAD

**Discussion by SpandexWizard** [vroland/epdiy#323] [General]
what sort of power consumption does the epdiy have? i'm trying to weigh my options, as yall have convinced me that trying to get it to run on just my pi might be a bad idea. i'm trying to make a portable device with as long a battery life as possible, so originally i was trying to remove the esp32 from the equation.

also does the firmware handle things like shutting down the wifi/bluetooth components? i definitely dont need wireless capability on my screen display module, so i have to assume shutting it off would save power

**Comment by mickeprag** [comment]
Epdiy is just a library, it does image rendering. The other stuff like wifi is up to you.
If you don't need wifi then esp-idf will leave it off by default, so just don't turn it on... ;)

If your goal is to use as little power as possible I would recommend you doing the other way around. Get rid of the power hungry Pi and keep the esp32.

**Reply by SpandexWizard** [reply]
if the esp32 is capable of outputting 2400x1034 resolution (preferably with the ability to remember screenspace so i can try some fast updating tricks), handling a probably simple typing program, and has enough juice to also BE the keyboard, then maybe i can pivot away from the pi.  but it's my understanding that the esp32 will struggle just to keep a copy of the screenspace in memory? either way this is why i want data. there's so much i dont know yet, and some things are just not out there. like how much power the epdiy board draws. i cant calculate how long my battery will last or how large a battery i even need.

**Comment by martinberlin** [comment]
If you want I can upload some reports about the v7 board consumption when updating a 9.7" panel. I will just take the measurements from 5V USB with Nordik PPK. It might be a bit different when being powered by a 4 volt Lipo battery but for a reference it might work.
On updates the Eink's generally consume much more than a LCD. They are low consumption because of the fact they retain the last image, but as you can see in the Schematics, they need different voltages generated by a Power IC. Will check my measurements in the past and also maybe make a blog-post about it

**Reply by SpandexWizard** [reply]
some data is better than no data! a baseline definitely helps, i'd really appreciate it. esp32's have a reputation in the mechanical keyboard community for sipping power because they have rather aggressive sleep capabilities, so i'm curious if that sort of thing is being taken advantage of in the epdiy firmware?

**Reply by martinberlin** [reply]
Didn’t forgot about this. Over the weekend I’m connecting one of my v7 boards to 4V with Nordic PPK and adding some consumption reports in the WiKi

**Comment by martinberlin** [comment]
@SpandexWizard will be start slowly adding consumption tests here:

https://github.com/vroland/epdiy/wiki/Board-v7-Consumption

More deepsleep consumption will come later over the week. But basically when it updates the display it consumes an AVG of 140mA. That's why it's important to use MODE_DU only monochrome to update fast and just go to sleep if you care about making a very low consumption device that needs to last long using a LiPo battery.

**Reply by SpandexWizard** [reply]
nods. thank you for putting this together, i really appreciate it!
